### ISSUE TITLE
Market

### ISSUE THREAD

**Issue by itajaja** [itajaja/sco#4]

**Comment by Kimerao0** [comment]
Il mercato serve ai giocatori per scambiare risorse, per fare investimenti e come fonte di denaro per i prestiti.

Mercato base: giacché il mercato è usato principalmente per lo scambio delle risorse questo parte ad inizio partita con una grande quantità di risorse (poniamo per esempio 100 volte la somma delle risorse iniziali dei giocatori). Ogni volta che un giocatore commercia col mercato le risorse del mercato variano di conseguenza, creando una variazione dei prezzi. 
Il prezzo di una risorsa è dato dalla disponibilità del mercato e si calcola come R1/R2 (dove R1 è la risorsa che si vuole vendere e R2 quella che si vuole comprare).
La schermata del mercato dovrebbe poter permettere ai giocatori di vedere, oltre ai prezzi correnti, l’andamento di ogni risorsa (calcolato come quantità di quella risorsa presente nel mercato al variare dei turni). I giocatori più esperti possono sfruttare l’andamento di mercato per investire in risorse di cui non hanno bisogno, ma che probabilmente guadagneranno potere d’acquisto nel prossimo futuro.
Gli scambi col mercato vengono svolti durante la preparazione dell’ordine d’esecuzione da parte del giocatore, ma il giocatore ha immediata disponibilità delle risorse scambiate. Durante la fase di aggiornamento verranno modificati i prezzi del mercato.
In questa forma il mercato non acquista risorse se non con gli scambi coi giocatori di conseguenza è possibile che il mercato esaurisca una o più risorse (logicamente viene da pensare che questa ipotesi sia alquanto improbabile dato che il considerevole aumento di valore della risorsa di cui il mercato scarseggia).

Tasse sugli scambi: si può impostare una tassa su ogni scambio, in questo modo si incentivano gli scambi di risorse fra giocatori. La tassa può essere imposta a partire da un determinato turno e può aumentare più avanti nel gioco. 

Prestiti: i giocatori possono farsi prestare delle risorse dal mercato, ad un tasso dipendente da vari fattori, entro un numero determinato di turni.
La risorsa chiesta in prestito non può eccedere la quantità prodotta in un turno dal richiedente ed il termine massimo è fissato in cinque turni. L’interesse base è del 16% per il primo turno di prestito, del 14% per il secondo, del 12% per il terzo, del 10% per il quarto e del 8% per il quinto turno. Questo vuol dire che se un giocatore chiede 100 oro di prestito da restituire in 3 turni dovrà ridare al mercato un totale di 142 (16+14+12) oro entro 3 turni.
Il giocatore dovrà restituire il prestito con la stessa risorsa, saranno quindi a suo carico eventuali tasse sugli scambi.
Se il prestito non dovesse essere restituito entro il termine il giocatore verrà penalizzato con un embargo, non potrà quindi più scambiare risorse col mercato fino al pagamento di una multa equivalente al 200% della somma prestata. Il giocatore dovrà restituire il prestito con la stessa risorsa, saranno quindi a suo carico eventuali tasse sugli scambi.

Investimenti: possiamo introdurre della azioni, tali azioni non produrranno dividendi, ma potranno portare vantaggi economici se acquistate e vendute in momenti propizi. L’andamento delle azione deriverà da delle variabili sconosciute al giocatore, ma intuibili. 
Per esempio le azioni di un’azienda che produce navi piccole aumenteranno di valore in base alla diffusione nel gioco di navi di questo tipo.

Vorrei poter introdurre anche la possibilità di investire su altri giocatori; questo avviene quando un giocatore decide di vendere titoli di stato del suo paese. Sarà il giocatore che vende ad imporre il prezzo del titolo e la quantità venduta. Gli altri giocatori possono comprare i titoli al prezzo imposto oppure non comprarli. Ogni titolo comprato abbasserà di uno il livello massimo di “controllo” del giocatore che vende. I titoli possono anche essere rivenduti al proprietario originale, il prezzo sarà fissato dal venditore e riporterà i punti di controllo al legittimo proprietario.

**Comment by theodred25** [comment]
I agree with the main idea of a Market which starts with a great number of resources and then sustains itself with the exchanged resources only (as I proposed the same thing on WhatsApp), although I'd have a hard time trying to justify this solution writing the game's background, especially if there is no equivalent to the current Confederation of United Planets which could be seen as the seller of resources. So, the reason I like this solution is just I couldn't think of another one which includes the main features of any market, namely that prices rise if a resource is scarce *and* if there is a great demand of it. This is why I don't think we should use the simple formula given above, in which the price depends only on the current availability of resources: even if what I sell is very rare, if few people are interested in it I must keep low prices in order to sell it. Thus, the formula should depend also on the amount of resources bought in the previous turn (or previous N turns if we have to smooth prices behavior).

Regarding the possibility of running out of a resource: as stated above the probability of such an event is low, since prices will rise with the depletion of that resource, to the point that buying it is not convenient anymore, while selling is. Also, tweaking the formula for prices, this probability can be made as low as one wants, since prices can be made rise faster as market is running out of resources. Anyway, this possibility must be taken into account; as I wrote on WhatsApp, a possible solution is as follows: if, during a turn, players buy more than what the market currently owns, purchases are made anyway, and the amount of that resource goes negative (this can be justified referring to an "emergency supply"); then, starting from the next turn, that resources cannot be bought anymore, until players sell enough of it to the market that it becomes positive again (or greater than some fixed quantity, if we want to make it harder for the resource to be sold out again).

Regarding taxes: their presence has no practical consequence designing the game, since when we agree on a formula for calculating prices, including taxes is just a matter of deciding a factor to multiply that formula with. For example, if taxation behaves like 20% VAT, you just have to multiply 1.2 and the original price.
